app.rb

```diff
@@ -180,6 +180,36 @@ class << self
     CategoryContribution.all.to_json
   end
 
+  get '/api/payments/?:id?' do |id|
```

Comment:
If you're using `id` in the where clause, the path should be `/api/payments/:id` as the id is not actually optional.